eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1428,6 +1428,16 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
       // -> in CLOSING we either have mutual closed (so no more htlcs), or already have unilaterally closed (so no action required), and we can't be in OFFLINE state anyway
       handleLocalError(HtlcTimedout(d.channelId, d.commitments.timedoutOutgoingHtlcs(count)), d, Some(c))
 
+    case Event(c@CurrentFeerates(feeratesPerKw), d: HasCommitments) =>
```

Comment:
This needs to also be done in state `SYNCING` (we can stay in that state a long time in some corner cases).

Can we also factorize in a `handleCurrentFeerate` handler?

Reply:
ACK for applying the logic in `SYNCING` too, factorizing the code in a handler is nice too but the fee-mismatch checks are slightly different between the normal case and `OFFLINE`/`SYNCING`. For example when `OFFLINE` we only want to fail the channel if the fees are raising over a certain threshold, as opposed to a more general mismatch used during `NORMAL` or `SHUTDOWN`

Reply:
Commit e98301 handles the feerates in `SYNCING` and refactored the feerate logic into 2 different handlers, one used during normal channel operation and the other used when the channel is not operational (`OFFLINE`/`SYNCING`)